app/src/main/java/com/nilhcem/blenamebadge/ui/badge_preview/PreviewBadge.kt

```diff
@@ -82,6 +84,51 @@ class PreviewBadge : View {
                 View.MeasureSpec.makeMeasureSpec(calculatedHeight, View.MeasureSpec.EXACTLY))
     }
 
+    public override fun onSaveInstanceState(): Parcelable? {
+        val bundle = Bundle()
+        bundle.putParcelable("superState", super.onSaveInstanceState())
+        bundle.putBoolean("ifFlash", this.ifFlash)
+        bundle.putBoolean("ifMarquee", this.ifMarquee)
+        bundle.putInt("badgeSpeed", this.badgeSpeed)
+        bundle.putInt("badgeMode", when (badgeMode) {
+            Mode.LEFT -> 1
+            Mode.RIGHT -> 2
+            Mode.UP -> 3
+            Mode.DOWN -> 4
+            Mode.FIXED -> 5
+            Mode.SNOWFLAKE -> 6
+            Mode.PICTURE -> 7
+            Mode.ANIMATION -> 8
+            Mode.LASER -> 9
+        })
+        bundle.putParcelableArrayList("checkList", ArrayList(this.checkList))
```

Comment:
Use putParcelable

Reply:
putParcelable wont work for Lists sir

Reply:
Why you want to store list of checklist?

Reply:
To check them with the cells when working with animations

Reply:
The Animations work by having two 2d lists, we have one as the base and another which checks if the index of the list has true or false, which makes the led glow. So, to create these animations we just play with the indexes of the lists, using some simple math operations the animations appear like they currently are

Reply:
I'll repeat, Why you want to store list of checklist and not just checklist? It makes no sense to put it in ArrayList and save in bundle

Reply:
@iamareebjamal Checklist contains a single list, we need a 2d list for the x and y coordinates. Therefore, a list of checklists

And we want to save it because it stores the binary values in the boolean of the the drawable or text

Reply:
`bundle.putParcelableArrayList("checkList", this.checkList)`

Why can't you write this?

You are storing checkList after wrapping in ArrayList, and while restoring, you just assign

```
this.checkList = bundle.getParcelableList
```

Makes absolutely no sense to me. Either wrap and extract in both places or wrap nowhere